flake.nix

```diff
@@ -1,61 +1,36 @@
 {
   description = "nix.dev static website";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
   inputs.flake-utils.url = "github:numtide/flake-utils/master";
-  inputs.poetry2nix = {
-    inputs.flake-utils.follows = "flake-utils";
-    inputs.nixpkgs.follows = "nixpkgs";
-    url = "github:nix-community/poetry2nix/master";
-  };
 
-  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+  outputs = { self, nixpkgs, flake-utils}:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ poetry2nix.overlay ];
         };
-        poetryOverrides = self: super: {
-
-          # Refs https://github.com/nix-community/poetry2nix/issues/218#issuecomment-981615612
-          typing-extensions = super.typing-extensions.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
-
-          sphinx-design = super.sphinx-design.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
+      in {
+        defaultPackage = pkgs.stdenv.mkDerivation {
+          name = "nix-dev";
+          src = self;
+          buildInputs = with pkgs.python310Packages; [
```

Comment:
```suggestion
          nativeBuildInputs = with pkgs.python310Packages; [
```

Reply:
Can you explain the motivation here? I gather that (native|propagated)BuildInputs and buildInputs are rooted in cross compilation, but this distinction is lost on me in the context of Python.

Reply:
Simply correctness; native build inputs are your tools. Yes, outside cross-compilation it is less valuable however for Python it for sure still matters. Remember, there is a lot of native code used by Python modules.

Note `buildPython*` makes the distinction (it sets `strictDeps = true;`).

Reply:
Let me rephrase the question to hone in on the part that confuses me: in general and in the Python specific case, when should I put packages in buildInputs, nativeBuildInputs, or propagatedBuildInputs?

Reply:
- `nativeBuildInputs`: these are build-time tools, that is, programs that run *during* the build (hence they need to be native)
- `buildInputs`: these are headers/libraries which one needs during build time, e.g. to link against, but it is not executed. These are thus your run-time dependencies.
- `propagatedBuildInputs`: these are `buildInputs` but that are propagated, that is, whoever adds your derivation in a `buildInputs`, will automatically get all these `propagatedBuildInputs` as well. And this is done recursively.

The Nixpkgs manual describes this in more detail.

In `buildPython*` we use `propagatedBuildInputs` for our run-time dependencies, which in a way is in line with what I wrote above, but it has its issues. It's a historic artifact and unfortunately a complete pain to fix.